components/rdf-update.js

```diff
@@ -4,65 +4,51 @@ var _ = require('underscore');
 var Promise = require('promise');
 var Handlebars = require('handlebars');
 var rdfstore = require('rdfstore');
-var noflo = require('noflo');
 
 var basenode = require('./base-node');
+var promiseComponent = require('./promise-component');
 
-exports.getComponent = function() {
-    return _.extend(new noflo.Component({
-        outPorts: {
-            out: {
-                description: "RDF JS Interface Graph object",
-                datatype: 'object'
-            },
-            error: {
-                description: "Error object",
-                datatype: 'object'
-            }
+exports.getComponent = promiseComponent({
+    description: "Executes the given SPARQL update on the provided RDF graph and returns it",
+    icon: 'cogs',
+    inPorts: {
+        parameters: {
+            description: "A map of template parameters",
+            datatype: 'object',
+            ondata: basenode.assign('parameters')
         },
-        inPorts: {
-            parameters: {
-                description: "A map of template parameters",
-                datatype: 'object',
-                process: basenode.on({data: basenode.assign('parameters')})
-            },
-            update: {
-                description: "SPARQL update template string in handlebars syntax",
-                datatype: 'string',
-                required: true,
-                process: basenode.on({data: basenode.assign('update', Handlebars.compile)})
-            },
-            "default": {
-                description: "Graph URI for the default dataset",
-                datatype: 'string',
-                process: basenode.on({data: basenode.assign('defaultURIs', basenode.push)})
-            },
-            namespace: {
-                description: "Graph URI for the named dataset",
-                datatype: 'string',
-                process: basenode.on({data: basenode.assign('namespaceURIs', basenode.push)})
-            },
-            'in': {
-                description: "RDF JS Interface Graph object",
-                datatype: 'object',
-                required: true,
-                process: basenode.on({data: execute})
-            }
+        update: {
+            description: "SPARQL update template string in handlebars syntax",
+            datatype: 'string',
+            required: true,
+            ondata: basenode.assign('update', Handlebars.compile)
+        },
+        "default": {
+            description: "Graph URI for the default dataset",
```

Comment:
The default Graph (like in a FROM clause) https://www.w3.org/TR/sparql11-query/#unnamedGraph